datadog/tests/datadog_test.go

```diff
@@ -19,10 +20,109 @@ import (
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
-// this test gets DD keys from env vars, writes a config file
-// then it invokes the plugin and validates some basics
-// in the response
-func TestDDCostRetrieval(t *testing.T) {
+func TestDDCostRetrievalListCost(t *testing.T) {
+	// query for qty 2 of 1 hour windows
+	windowStart := time.Date(2024, 3, 8, 0, 0, 0, 0, time.UTC)
+	windowEnd := time.Date(2024, 3, 8, 2, 0, 0, 0, time.UTC)
+
+	response := getResponse(t, windowStart, windowEnd, time.Hour)
+
+	// confirm no errors in result
+	if len(response) == 0 {
+		t.Fatalf("empty response")
+	}
+	for _, resp := range response {
+		if len(resp.Errors) > 0 {
+			t.Fatalf("got errors in response: %v", resp.Errors)
+		}
+	}
+
+	result, err := json.MarshalIndent(response, "", "    ")
+	if err != nil {
+		t.Fatalf("error json-ing response: %v", err)
+	}
+
+	fmt.Println(string(result))
+	// confirm results have correct provider
+	for _, resp := range response {
+		if resp.Domain != "datadog" {
+			t.Fatalf("unexpected domain. expected datadog, got %s", resp.Domain)
+		}
+	}
+
+	// check some attributes of the cost response
+	for _, resp := range response {
+		// confirm there are > 0 custom costs
+		if len(resp.Costs) < 1 {
+			t.Fatalf("expect non-zero costs in response.")
+		}
+
+		for _, cost := range resp.Costs {
+			if cost.ResourceType == "indexed_logs" {
+				// check for sane values fo a rate-priced resource
+				if cost.ListCost > 1000 {
+					spew.Dump(
+						cost,
+					)
```

Comment:
Same deal here -- you can use `spew.Sdump` to get a string to then log.

Reply:
Neat! That's good to know.